lib/sensors/include/UltraSonicSensor.h

```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include "Pwmln.h"
+#include "mbed.h"
+
+// declare constants
+const int NUM_SAMPLES{1};
+const int CONV_FACTOR{1000000};
+
+class UltraSonicSensor {
+ public:
+  // Constructor for UltrasonicSensor
+  UltraSonicSensor(PinName echoPin, PinName trigPin);
+  float getDistance();
+  uint64_t pulseIn();
+
+ private:
+  InterruptIn _echoPin;
+  DigitalOut _trigPin;
+  // enum class to contain states of echoPin
+  class enum PinState { PIN_LOW = 0, PIN_HIGH = 1 };
```

Comment:
unnecessary imo. As seen [here](https://os.mbed.com/docs/mbed-os/v6.3/apis/digitalout.html), mbed overloads the `=` operator of digitalout pins. throughout our codebase, i believe we use this `pin_name = 1;` paradigm to turn pins on.